lib/AST/ASTMangler.cpp

```diff
@@ -4009,6 +4016,11 @@ std::string ASTMangler::mangleMacroExpansion(
       expansion->getMacroName().getBaseName().userFacingName(),
       MacroRole::Declaration,
       expansion->getDiscriminator());
+  auto privateDiscriminator = getPrivateDiscriminatorIfNecessary(expansion);
+  if (!privateDiscriminator.empty()) {
+    appendIdentifier(privateDiscriminator);
+    appendOperator("Ll");
+  }
```

Comment:
This looks like an extension to the mangling grammar. Can you document the mangling change in `Mangling.rst` and make sure that these mangled names are demangled appropriately (you can add one of the mangled names produced by your tests to `test/Demangle/Inputs/manglings.txt` along with the printed result). I'm not sure if you'll need to update the demangle tree/demangler/remanglers or not, but that's the way to find out.

Reply:
I think it's already covered by the existing grammar. In Mangling.rst have

```
decl-name ::= identifier identifier 'LL'            // file-discriminated declaration
```

Reply:
Ah I see that macro entities don't have a `decl-name`. I can add it separately to the `macro-discriminator-list` rule.

Reply:
Done.